### ISSUE TITLE
Compiler code missing?

### ISSUE THREAD

**Issue by jovanbulck** [SSGAalto/sgx-branch-shadowing-mitigation#1]
Dear authors,

Thanks for open-sourcing the code accompanying your SysTEX paper! This is an interesting project and I'm trying to better understand the LLVM mitigation. However, as there's no commit history or git submodules, I had some troubles trying to isolate the compiler changes and figure out on which exact LLVM version/commit the defense is implemented. After a lot of semi-automated and manual poking around, I think I managed to establish that the llvm directory is based on an upstream commit from  Apr 11, 2018:

https://github.com/llvm-mirror/llvm/commit/e7bb73e01b10919f2312a2b4ae734d8be4ee6576

However, when doing a git diff on a freshly checked out repository, it seem that the llvm subdirectory in this project is an _exact_ copy of the above commit, without the actual modifications to defend against branch shadowing attacks? Could it be that you forgot to push the actual defense compiler pass, or am I missing something and could you point me to the LLVM files implementing the mitigation?

Note that it would generally be better to add the mitigation changes as a separate commit so one can easily see the changes separately, and/or use git [submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) with separate patch files. I attempted to start doing that here, but afaik the actual LLVM code implementing the mitigation is still missing:

https://github.com/jovanbulck/sgx-branch-shadowing-mitigation

Thanks again for the interesting work!
Jo

**Comment by ishkamiel** [comment]
Your suspicion is correct, we have indeed forgot to push in our actual changes. Thank you for letting us know and for the interest in our work.

I've now pushed in the changes to the master branch. I've also pushed in a new branch (https://github.com/SSGAalto/sgx-branch-shadowing-mitigation/tree/shadow-llvm) which is a fork from the release_60 branch from the old LLVM SVN git mirror (based on commit https://github.com/llvm-mirror/llvm/commit/d88ca2927027d254931437c6f6c83f43724945af)

I'm sorry for any wasted time due to our mistake, and thank you for informing us!
-hans
